### PR TITLE
Split factories

### PR DESCRIPTION
**Pull request by PierreMesure**
References
===================
#2826

Objectives
===================
Split the factories.rb file into smaller logical files

Visual Changes
===================
None

Notes
===================
- Not too sure on *campaigns*, *i18n_content*, *identity* and a few others. Feedback welcome

### CODE REVIEW

File: spec/factories/administration.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  factory :widget_feed, class: 'Widget::Feed' do
+  end
+
+  factory :i18n_content, class: 'I18nContent' do
+    key 'debates.index.section_footer.description'
+    value_es 'Texto en español'
+    value_en 'Text in english'
+  end
+end
```
Comment (aitbw) [inline spec/factories/administration.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/analytics.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  factory :visit  do
+    id { SecureRandom.uuid }
+    started_at { DateTime.current }
+  end
+
+  factory :campaign do
+    sequence(:name) { |n| "Campaign #{n}" }
+    sequence(:track_id) { |n| n.to_s }
+  end
+end
```
Comment (aitbw) [inline spec/factories/analytics.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/budgets.rb [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    association :status, factory: :budget_investment_status
+    sequence(:title)     { |n| "Budget investment milestone #{n} title" }
+    description          'Milestone description'
+    publication_date     { Date.current }
+  end
+
+  factory :valuator_group, class: ValuatorGroup do
+    sequence(:name) { |n| "Valuator Group #{n}" }
+  end
+end
```
Comment (aitbw) [inline spec/factories/budgets.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/classifications.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+  factory :topic do
+    sequence(:title) { |n| "Topic title #{n}" }
+    sequence(:description) { |n| "Description as comment #{n}" }
+    association :author, factory: :user
+  end
+
+  factory :related_content do
+  end
+end
```
Comment (aitbw) [inline spec/factories/classifications.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/comments.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      valuation true
+      association :commentable, factory: :budget_investment
+      before :create do |valuation|
+        valuator = create(:valuator)
+        valuation.author = valuator.user
+        valuation.commentable.valuators << valuator
+      end
+    end
+  end
+end
```
Comment (aitbw) [inline spec/factories/comments.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/debates.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        4.times { create(:vote, votable: debate) }
+      end
+    end
+  end
+
+  factory :flag do
+    association :flaggable, factory: :debate
+    association :user, factory: :user
+  end
+end
```
Comment (aitbw) [inline spec/factories/debates.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/emails.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :newsletter do
+    sequence(:subject) { |n| "Subject #{n}" }
+    segment_recipient  UserSegments::SEGMENTS.sample
+    sequence(:from)    { |n| "noreply#{n}@consul.dev" }
+    sequence(:body)    { |n| "Body #{n}" }
+  end
+end
```
Comment (aitbw) [inline spec/factories/emails.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/files.rb [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+      resource_relation "documents"
+      attachment { File.new("spec/fixtures/files/empty.pdf") }
+    end
+    trait :image do
+      resource_relation "image"
+      attachment { File.new("spec/fixtures/files/clippy.jpg") }
+    end
+    initialize_with { new(attributes) }
+  end
+end
```
Comment (aitbw) [inline spec/factories/files.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/files.rb [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+      resource_type "Proposal"
+    end
+    trait :budget_investment do
+      resource_type "Budget::Investment"
+    end
+
+    trait :documents do
+      resource_relation "documents"
+      attachment { File.new("spec/fixtures/files/empty.pdf") }
+    end
```
Comment (aitbw) [inline spec/factories/files.rb]: Can you add a white line **right after** this `end`?

File: spec/factories/files.rb [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    trait :image do
+      resource_relation "image"
+      attachment { File.new("spec/fixtures/files/clippy.jpg") }
+    end
```
Comment (aitbw) [inline spec/factories/files.rb]: Can you add a white line **right after** this `end`?

File: spec/factories/legislations.rb [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+  end
+
+  factory :legislation_proposal, class: 'Legislation::Proposal' do
+    title "Example proposal for a legislation"
+    summary "This law should include..."
+    terms_of_service '1'
+    process factory: :legislation_process
+    author factory: :user
+  end
+end
```
Comment (aitbw) [inline spec/factories/legislations.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/notifications.rb [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    segment_recipient UserSegments::SEGMENTS.sample
+    recipients_count  nil
+    sent_at           nil
+
+    trait :sent do
+      recipients_count 1
+      sent_at Time.current
+    end
+  end
+end
```
Comment (aitbw) [inline spec/factories/notifications.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/organizations.rb [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+    trait :verified do
+      verified_at { Time.current }
+    end
+
+    trait :rejected do
+      rejected_at { Time.current }
+    end
+  end
+end
```
Comment (aitbw) [inline spec/factories/organizations.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/polls.rb [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+    association :officer, factory: :poll_officer
+    document_number
+    document_type    "1"
+    year_of_birth    "1980"
+
+    trait :invalid do
+      year_of_birth { Time.current.year }
+    end
+  end
+end
```
Comment (aitbw) [inline spec/factories/polls.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/proposals.rb [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+    signature_sheet
+    sequence(:document_number) { |n| "#{n}A" }
+  end
+
+  factory :activity do
+    user
+    action "hide"
+    association :actionable, factory: :proposal
+  end
+end
```
Comment (aitbw) [inline spec/factories/proposals.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/users.rb [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+    end
+  end
+
+  factory :direct_message do
+    title    "Hey"
+    body     "How are You doing?"
+    association :sender,   factory: :user
+    association :receiver, factory: :user
+  end
+end
```
Comment (aitbw) [inline spec/factories/users.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/verifications.rb [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+    user
+    tries 0
+    locked_until { Time.current }
+  end
+
+  factory :verified_user do
+    document_number
+    document_type 'dni'
+  end
+end
```
Comment (aitbw) [inline spec/factories/verifications.rb]: Please, add a trailing white line at the end of this file :pray:

File: spec/factories/votes.rb [hunk 18]
```diff
@@ -0,0 +1,11 @@
...
+  factory :vote do
+    association :votable, factory: :debate
+    association :voter,   factory: :user
+    vote_flag true
+    after(:create) do |vote, _|
+      vote.votable.update_cached_votes
+    end
+  end
+end
+
```
Comment (aitbw) [inline spec/factories/votes.rb]: This one has an extra trailing white line, can you please remove it? :pray:

**Comment by PierreMesure** [comment]
Done! Sorry, my editor was badly configured again.

**Comment by voodoorai2000** [comment]
Wonderful! Thank you so much @PierreMesure 🙌

**Pull request merged**